tests/async/test_video.py

```diff
@@ -49,3 +49,15 @@ async def test_short_video_should_throw_persistent_context(
 
     path = await page.video.path()
     assert str(tmpdir) in str(path)
+
+
+async def test_should_not_error_if_page_not_closed_before_save_as(
+    browser, tmpdir, server
+):
+    page = await browser.new_page(record_video_dir=tmpdir)
+    await page.goto(server.PREFIX + "/grid.html")
+    out_path = tmpdir / "some-video.webm"
+    saved = page.video.save_as(out_path)
+    await page.close()
+    await saved
+    os.path.exists(out_path)
```

Comment:
```suggestion
    assert os.path.exists(out_path)
```

Reply:
updated (also added corresponding check in the error case, and then an assurance we can still save after)